server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadataVerifier.java

```diff
@@ -92,7 +89,7 @@ public IndexMetadata verifyIndexMetadata(IndexMetadata indexMetadata, Version mi
         // Next we have to run this otherwise if we try to create IndexSettings
         // with broken settings it would fail in checkMappingsCompatibility
         newMetadata = archiveBrokenIndexSettings(newMetadata);
-        createAndValidateMapping(newMetadata);
+        checkMappingsCompatibility(newMetadata);
```

Comment:
The changes in this class revert the changes made in https://github.com/elastic/elasticsearch/pull/85059, as we now validate against a MapperService with all analyzers configured when restoring the legacy index in `RestoreService`. The reason for doing it this way now is that it provides better error messages on restore, but also handles a tricky situation where the Mapping returned by these methods here would have their analyzer settings misconfigured as checkMappingsCompatibility would not create a proper environment with actual analyzers configured.